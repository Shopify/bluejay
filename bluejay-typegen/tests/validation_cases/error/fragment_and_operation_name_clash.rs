#[bluejay_typegen::typegen([
    type Query {
        myType: MyType
    }

    type MyType {
        field: String
    }
])]
mod schema {
    #[query([
        query Foo {
            myType {
                ...Foo
            }
        }

        fragment Foo on MyType {
            field
        }
    ])]
    mod query {}
}

fn main() {}
