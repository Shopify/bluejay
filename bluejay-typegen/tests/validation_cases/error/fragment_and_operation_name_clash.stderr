error: Error: bluejay_typegen requires fragment and operation names to be unique, but encountered a clash with name `Foo`
          ╭─[<unknown>:1:33]
          │
        1 │ query Foo { myType { ...Foo } } fragment Foo on MyType { field }
          │ ───────────────┬─────────────── ────────────────┬───────────────
          │                ╰────────────────────────────────────────────────── Operation definition
          │                                                 │
          │                                                 ╰───────────────── Fragment definition name collides with operation definition name
       ───╯
  --> tests/validation_cases/error/fragment_and_operation_name_clash.rs:11:13
   |
11 |       #[query([
   |  _____________^
12 | |         query Foo {
13 | |             myType {
14 | |                 ...Foo
...  |
20 | |         }
21 | |     ])]
   | |_____^
